node_lambnik/src/tiler/src/tiler.js

```diff
@@ -17,21 +17,42 @@ const TILE_WIDTH = 256
 // Register plugins
 mapnik.register_default_input_plugins()
 
+/**
+ * Disables all layers that aren't specified in the query string
+ */
+const filterLayers = (xmlString, layers) => {
+    // Only filter if there ARE layers to filter on
+    if (layers && layers.length > 0) {
+        return xmlString.replace(
+            /(<Layer name="(.+)")>/g,
+            (fullMatch, beginning, layerName) => {
+                if (!layers.includes(layerName)) {
+                    return `${beginning} status="0">`
+                }
+                return fullMatch
+            },
+        )
+    }
+
+    return xmlString
+}
+
 /**
  * Creates a map based on configured datasource and style information
  * @param z
  * @param x
  * @param y
  * @returns {Promise<mapnik.Map>}
  */
-const createMap = (z, x, y) => {
+const createMap = (z, x, y, layers) => {
     // Create a webmercator map with specified bounds
     const map = new mapnik.Map(TILE_WIDTH, TILE_HEIGHT)
     map.bufferSize = 64
     map.extent = new SphericalMercator().bbox(x, y, z, false, process.env.EPSG)
 
     // Load map specification from xml string
     return readFile(path.join(__dirname, 'map-config.xml'), 'utf-8')
+        .then(xml => filterLayers(xml, layers))
```

Comment:
Is it really true that layers can't be dropped after the map object is made?  It certainly [looks that way](http://mapnik.org/documentation/node-mapnik/3.6/#Map), but it seems like such obviously useful functionality...

Reply:
Yeah, I agree, it seems like a weird design choice to me. [I found this issue in node-mapnik](https://github.com/mapnik/node-mapnik/issues/742) where someone expresses a need for the same functionality, and Dane Springmeyer (who's the chief contributor to the project) said the following:
> This is a known limitation that I don't think should be changed. I don't think an API that allows mutating maps in place is within scope. I recommend thinking about map instances as best created once and then rendered one or more times and never modified again.